x/metadata/types/specification.go

```diff
@@ -0,0 +1,155 @@
+	}
+	for i, contractSpecID := range s.ContractSpecIds {
+		prefix, err = VerifyMetadataAddressFormat(contractSpecID)
+		if err != nil {
+			return fmt.Errorf("invalid contract specification id at index %d: %w", i, err)
+		}
+		if prefix != PrefixGroupSpecification {
+			return fmt.Errorf("invalid contract specification id prefix at index %d (expected: %s, got %s)",
+				i, PrefixGroupSpecification, prefix)
+		}
+	}
+	return nil
+}
+
+// String implements stringer interface
+func (s ScopeSpecification) String() string {
+	out, _ := yaml.Marshal(s)
+	return string(out)
+}
+
+// NewDescription creates a new Description instance.
+func NewDescription(name, description, websiteURL, iconURL string) *Description {
+	return &Description{
+		Name:        name,
+		Description: description,
+		WebsiteUrl:  websiteURL,
+		IconUrl:     iconURL,
+	}
+}
+
+// ValidateBasic performs basic format checking of data in an Description.
+// The path parameter is used to provide extra context to any error messages.
+// e.g. If the name field is invalid in this description, and the path provided is "ScopeSpecification.Description",
+// the error message will contain "ScopeSpecification.Description.Name" and the problem.
+// Provide "" if there is no context you wish to provide.
+func (d *Description) ValidateBasic(path string) error {
+	if len(d.Name) == 0 {
+		return fmt.Errorf("description %s cannot be empty", makeFieldString(path, "Name"))
+	}
+	if len(d.Name) > maxDescriptionNameLength {
+		return fmt.Errorf("description %s exceeds maximum length (expected <= %d got: %d)",
+			makeFieldString(path, "Name"), maxDescriptionNameLength, len(d.Name))
+	}
+	if len(d.Description) > maxDescriptionDescriptionLength {
+		return fmt.Errorf("description %s exceeds maximum length (expected <= %d got: %d)",
+			makeFieldString(path, "Description"), maxDescriptionDescriptionLength, len(d.Description))
+	}
+	err := validateURLBasic(d.WebsiteUrl, false, path, "WebsiteUrl")
+	if err != nil {
+		return err
+	}
+	err = validateURLBasic(d.IconUrl, false, path, "IconUrl")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// String implements stringer interface
+func (d Description) String() string {
+	out, _ := yaml.Marshal(d)
+	return string(out)
+}
+
+// validateURLBasic - Helper function to check if a url string is superficially valid.
+// The path and fieldName parameters are combined using makeFieldString for error messages.
+func validateURLBasic(url string, required bool, path string, fieldName string) error {
+	if len(url) == 0 {
+		if required {
+			return fmt.Errorf("url %s cannot be empty", makeFieldString(path, fieldName))
+		}
+		return nil
+	}
+	if len(url) > maxURLLength {
+		return fmt.Errorf("url %s exceeds maximum length (expected <= %d got: %d)",
+			makeFieldString(path, fieldName), maxURLLength, len(url))
+	}
+	if strings.ToLower(url[0:8]) != "https://" && strings.ToLower(url[0:7]) != "http://" {
```

Comment:
I wonder about this restriction ... seems like it might be nice to support data urls too...
```
data:image/png;base64,iVBORw0KGgoAAAANSU[...]VORK5CYII=
```